### ISSUE TITLE
Social networks field on user profiles

### ISSUE THREAD

**Issue by JohnXLivingston** [Chocobozzz/PeerTube#5947]
### Describe the problem to be solved

As said in https://github.com/Chocobozzz/PeerTube/issues/5429 , Mastodon can mark a link in user profiles as verified if the distant page point to the user profile with a "a" tag having a `rel="me"` attribute.
This is very usefull to authenticate users on the fedivers.

Peertube users should be able to add one or many links to their websites and social networks.
These links would be shown on the user's page, and could be used to verify links on Mastodon-like profiles.
They would also be added to the AP user informations.

Peertube could also implement the same verifying procedure (check that the remote page has a link to the user profile, with `rel="me"`). 

### Describe the solution you would like

In your account options, their would be a given number of fields to add external links.
Such field could be doubled with a `description` field where the user could add a label to the link.

On the user's profile, these links would be shown, with a `rel="me"` attribute.

Note: Mastodon does not render the Javascript when verifying remote links. So Peertube should also add these links in the backend generated DOM when accessing a user's page.

**Comment by FediVideos** [comment]
Just had user request this via FediTips, they have linked to their PeerTube account and would like it to turn green on their Mastodon account. Would be great if there was some kind of dedicated part of the PeerTube profile that includes rel=me

**Comment by DarthChillash** [comment]
I have been wrecking my brain over this. Please implement if you can.

**Comment by markwyner** [comment]
Please, please, fix this. In a PeerTube profile you can use the rel="me" attribute. However, PeerTube automatically overrides this by adding two additional classes.

In my profile, for example, I use the following:

`<a href="https://mas.to/@markwyner" rel="me">Mastodon</a>`

But it renders like this in the PeerTube UI:

`<a href="https://mas.to/@markwyner" rel="me noopener noreferrer" target="_blank">Mastodon</a>`

The "noopener" and "noreferrer" classes render the rel attribute moot, as Mastodon won't read it. I'd _love_ to see this behavior remedied.

Thanks so much!

**Comment by Chocobozzz** [comment]
@markwyner Is it not a Mastodon issue? I think it should be able to understand that one of the rel attribute is `me` :think

**Comment by DarthChillash** [comment]
Mark is mistaken. When we put rel="me" attribute in our channel description it's not actually visible from source. For example do a wget https://tux-edu.tv/c/lxr/ and examine it. There is no mention of this attribute, but if you inspect my page with F12 tools in a web browser then you will see where the attribute is.

I think this is where the challenge lies.

**Comment by DarthChillash** [comment]
Personally I edited my PeerTube page in its source code so it's now green and verified on my Mastodon profile. But only the instance admin can hack it like this. I would love to see this implemented properly.

**Comment by markwyner** [comment]
@DarthChillash I don’t understand how I’m mistaken. I know that the source doesn’t render. I analyzed the rendered link by using developer tools. I live there when I’m coding. But Mastodon should be able to render the link as part of the rendered code. Search engines [manage this just fine](https://duckduckgo.com/?q=mark+wyner+northtube).

But this sounds like a complicated recipe overall.

By definition, the attribute `noreferrer` [may have an impact](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/noreferrer) on how Mastodon interprets the link. So even if Mastodon is reading the javascript-rendered source, the `noreferrer` value may be the issue. Otherwise, as @Chocobozzz mentioned, it should be smart enough to parse multiple values in the `rel` attribute.

Is there a reason PeerTube requires Javascript to render pages? Someone at PeerTube who made this decision has their reasons, but this seems short sighted.

As an aside, admin customization is definitely not an option. Most people using PeerTube are not admins.

**Comment by markwyner** [comment]
> I think it's a pity (even if this problem isn't technically critical) that a Peertube page can't appear as verified on Mastodon … because Peertube, just like Mastodon, is in the Fediverse. On my Mastodon profiles, all my links with the “rel” attribute are displayed in green: my blog, my GitHub page, my OSM page, my Pixelfed and Flickr pages … EXCEPT Peertube.

@Jerome-Herbinet that's what confuses me most. One of the main reasons I moved to PeerTube was because I had an expectation of harmony with Mastodon. This one thing is kind of a big flaw, given that (reasonable) expectation. I hope they can figure it out.

**Comment by Chocobozzz** [comment]
FYI we fixed the link not properly rendered by PeerTube in https://github.com/Chocobozzz/PeerTube/commit/cf921354a9f07c004650e4bce0d2ba0d20d70049 so Mastodon should be able to verify your identify from a PeerTube platform.

(will be included in v7.1)

**Comment by markwyner** [comment]
> FYI we fixed the link not properly rendered by PeerTube in [cf92135](https://github.com/Chocobozzz/PeerTube/commit/cf921354a9f07c004650e4bce0d2ba0d20d70049) so Mastodon should be able to verify your identify from a PeerTube platform.
> 
> (will be included in v7.1)

Yay! Thanks so much!

**Comment by frankstrater** [comment]
@Chocobozzz As there are 2 options for link verification through Mastodon, would you consider adding the meta tag option and make it configurable through the admin settings? 

Similar to the Twitter/X thing in Settings > Configuration > Basic for Open Graph purposes

See:

```html
<link href="https://social.example.com/@username" rel="me">
```

https://docs.joinmastodon.org/user/profile/#verification

**Comment by markwyner** [comment]
> As there are 2 options for link verification through Mastodon, would you consider adding the meta tag option and make it configurable through the admin settings?

That works if you're an admin. But I wouldn't want to rely on my server's admin to make this happen. They may not even want that. Having the link attribute is the most accessible. Both options, of course, are a bonus.

**Comment by frankstrater** [comment]
Good point. It depends on what you intend to claim as "ownership". Usually rel="me" is used to link social media acounts (Twitter, Mastodon) to other personal accounts (Pixelfed, GitHub, etc.) or personal websites (blogs).

For PeerTube it would be nice if you can link user accounts, channels, videos, etc. but it feels a bit messy if it's dependent on markdown in descriptions. 

I still prefer using the HTML meta code (visible in view source, not in virtual dom easier to "discover") and maybe make it dynamic, so the instance can be linked by admin and PeerTube "pages" (user, channel, video, etc.) can be linked by to a provided URL through User > Settings and Channel > Settings, etc.

I'll give it some more thought and will start working on it (upstream or plugin).

**Comment by frankstrater** [comment]
Proof of concept (hacked into HTML template):

https://mastodon.social/@frankstrater
https://peertube.beeldengeluid.nl/ (view source)
